src/rpc.rs

```diff
@@ -20,18 +20,42 @@ pub struct SubmitTransactionRequest {
     pub transaction: Transaction
 }
 
+/// Submit a transaction to the mempool.
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub struct SubmitTransactionRequestV2 {
+    pub transaction: TransactionV1OrV2
+}
+
 #[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
 pub struct SubmitTransactionResponse {
     pub error: Option<SubmitTransactionError>,
 }
 
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub struct SubmitTransactionResponseV2 {
+    pub error: Option<SubmitTransactionErrorV2>,
+}
+
 #[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
 pub enum SubmitTransactionError {
     UnacceptableNonce,
     MempoolFull,
     Other,
 }
 
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub enum SubmitTransactionErrorV2 {
```

Comment:
It seems like there is still [some debate](https://github.com/parallelchain-io/pchain-mempool/pull/2#discussion_r1324003431) about whether we should replace an existing transaction in the mempool if a different transaction with the same signer and nonce is received.

I will extend the set of variants in this enum (and in the other “Reason” enums) in the v0.5 protocol spec assuming that we will replace existing transactions.